base/point.h

```diff
@@ -25,13 +25,19 @@ class Point {
    * \brief Standard destructor.
    */
   virtual ~Point();
-  /** General accessor method for the underlying data.
+
+  /** General accessor method for each feature of the underlying data.
    * \param feature_id The coordinate of interest. Indexing is implementation
    * specific.
    * \return The value of the coordinate specified in the input.
    */
   virtual double GetFeatureValue(unsigned feature_id) const = 0;
 
+  /** General accessor method for the label of the underlying data.
+   * \return The label of the object.
+   */
+  virtual double GetLabel() const = 0;
```

Comment:
Regarding PredictProbabilty of the learner interface. I inspired myself in Weka Classifier abstract class that is used for both classification and regression problems, however, when they implement a Regressor they ignore the method of probability. how would you suggest to define the Learner interface?

Regarding the Label. It would be suitable for both, regression and classification problems if it was a double/float type. I agree that it can be ignored for unsupervised learning if we want a pointer that does not waste memory with it we just need to implement a Pointer for unsurpervised learning which implements the Point Interface.. dont you guys agree?